### DISCUSSION TITLE
Feature request for Scheduling Net Speed Indicator

### DISCUSSION THREAD

**Discussion by vpsj** [tribalfs/GalaxyMaxHzPub#172] [Ideas]
Hi, I love this app since it provides me my current net speed in the status bar without that annoying persistent notification like all the other similar apps do.

However, is it possible add a feature where a user can choose a specific time period where you want the indicator to be active? I searched in the app and I couldn't find any option to do so as of now. I tried a bixby routine to auto close the app at night but that didn't work either.

I have tested with this app installed and uninstalled, and in my (admittedly non-scientific) testing, it drains quite a fair bit of battery at night when the phone is just idle as I'm sleeping

The net speed indicator is the only feature I use from this app (everything else is left at default).

I'm not sure how the app works when the screen is off and/or when the phone is in doze mode, but would turning off the indicator when not required improve battery performance?

Please let me know if adding such a feature is feasible and would it even make a difference in the end or not. 

Thank you

Just in case - My device is S23 Ultra (latest update OneUI 6.1)

**Comment by tribalfs** [comment]
Is the phone connected to internet all night?

Actually this feature is designed to turn off automatically when disconnected to the internet and when the screen is off.
But let me investigate this .

Thank you.

**Reply by vpsj** [reply]
Yes, my phone is connected via Wifi 24/7 (although I don't know what happens to Wifi when the phone goes to deep doze). There are some specific automated tasks that my phone performs (like changing the color/brightness of my smart bulb) so I don't turn off my wifi at night.

Do you think the Wifi requirement is necessary for the speed indicator? From what I've seen, it's not logging the downloaded data or anything, it's just showing the realtime speed when the user is actually using their phone.. so if the phone screen is off, there is no need for the indicator to be on, right?

Either way, I'll be happy if it turns off when the screen is off... or if we can schedule an on/off timing for it. You'll be the best judge of which one would be better

Thank you for looking into this! :)

**Reply by tribalfs** [reply]
Yes, it turns off when screen is off and also when disconnected to the net. It does not log data usage - it just open the device built-in.
You can check [v8.1.50 update](https://github.com/tribalfs/GalaxyMaxHzPub/releases/tag/v8.1#:~:text=Latest%20Build%3A%20v8.1.50).
I added a few optimizations on it.
Thank you.
